[Docs] Add vizro-ai chart examples to docs (#469)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/chart-examples.md

> # Explore VizroAI

Reviewer: ```suggestion
# Gallery of examples
```

> In this guide, we'll learn how to make more advanced charts with Plotly using data from [Plotly Express](https://plotly.com/python-api-reference/generated/plotly.express.data.html). The examples below use the OpenAI `"gpt-4-0613"` model as we are going to request specific updates to the layout of the charts, which are [more complex than the default GPT 3.5 model can handle](./customize-vizro-ai.md).

Reviewer: ```suggestion
Take a look at some more advanced charts that can be created with Vizro-AI using data from [Plotly Express](https://plotly.com/python-api-reference/generated/plotly.express.data.html). The examples below use the OpenAI `"gpt-4-0613"` model as we are going to request specific updates to the layout of the charts, which are [more complex than the default GPT 3.5 model can handle](./customize-vizro-ai.md).
```

> ### Create a polar bar chart

Reviewer: ```suggestion
### Polar bar chart
```

A polar bar chart is a circular graph where each axis represents a different variable, typically used for displaying cyclical or directional data.
It's suitable for quickly comparing multiple variables across different categories or directions. Let's make one using Vizro-AI.

!!! example "Polar Bar Chart"

    === "Code for the cell"
        ```py
        import vizro_ai
        from vizro_ai import VizroAI
        import plotly.express as px

        from dotenv import load_dotenv
        load_dotenv()

        df = px.data.wind()

        vizro_ai = VizroAI(model="gpt-4-0613")
        vizro_ai.plot(df,
                      """Describe wind frequency and direction using bar_polar chart.
                         Increase the width and height of the figure.
                         Improve layout by placing title to the left. Show legend""", explain=True)

        ```
    === "Resulting chart"
        [![VizroAIChart1]][VizroAIChart1]

    [VizroAIChart1]: ../../assets/user_guides/polar_bar_chart.png

> ### Create a geographical map chart

Reviewer: ```suggestion
### Geographical map chart
```
or maybe even Choropleth chart?

The next chart we'll look at is a geographical map chart to visualize spatial patterns in data, which often reveals insights not seen in other charts.

!!! example "Map chart"

    === "Code for the cell"
        ```py
        import vizro_ai
        from vizro_ai import VizroAI
        import plotly.express as px

        from dotenv import load_dotenv
        load_dotenv()

        df = px.data.wind()

        vizro_ai = VizroAI(model="gpt-4-0613")
        vizro_ai.plot(df,
                      """Visualize life expectancy over the years using map chart. Use life expectancy as the color dimension.
                         Improve layout by using Arial font. Increase the width and height of the map area. Outline continents on the map.
                         Show countries on the map.
                         Increase the width and height of the figure.""", explain=True)

        ```
    === "Resulting chart"
        [![VizroAIChart2]][VizroAIChart2]

    [VizroAIChart2]: ../../assets/user_guides/map_chart.gif

> ### Create a surface plot

Reviewer: ```suggestion
### 3D surface plot
```

# Explore VizroAI

> In this tutorial, we walk through the process of creating more complex plotly charts. The examples use data from the [plotly express data package](https://plotly.com/python-api-reference/generated/plotly.express.data.html).
> In the examples below we will use OpenAI `"gpt-4-0613"` model as we are going to request specific updates to the layout of the charts.

Reviewer 2: ```suggestion
In this guide, we'll learn how to make more advanced charts with Plotly. We'll be using data from [Plotly Express](https://plotly.com/python-api-reference/generated/plotly.express.data.html) for our examples. Specifically, we'll be using the `"gpt-4-0613"` model from OpenAI to make specific layout changes to our charts.

```

> In the examples below we will use OpenAI `"gpt-4-0613"` model as we are going to request specific updates to the layout of the charts.

Reviewer 3: ```suggestion
The examples below use the OpenAI `"gpt-4-0613"` model as we are going to request specific updates to the layout of the charts, which are [more complex than the defaul GPT 3.5 model can handle](./customize-vizro-ai.md).
```

### Create polar bar chart with Vizro-AI

> Polar Bar Charts are mostly used to visualize data that has both magnitude and direction, such as wind speed and direction at specific locations.

Reviewer 3: ```suggestion
Polar bar charts are mostly used to visualize data that has both magnitude and direction, such as wind speed and direction at specific locations.
```

> They are particularly effective for displaying cyclical or directional data as they provide intuitive way of visualizing patterns in data across different directions.

Reviewer 3: ```suggestion
They are a particularly effective of displaying cyclical or directional data as they provide an intuitive way of visualizing patterns in data across different directions.
```

> ### Create polar bar chart with Vizro-AI
> 
> Polar Bar Charts are mostly used to visualize data that has both magnitude and direction, such as wind speed and direction at specific locations.
> They are particularly effective for displaying cyclical or directional data as they provide intuitive way of visualizing patterns in data across different directions.
> Let's create one using Vizro-AI.

Reviewer 2: ```suggestion
### Create a polar bar chart

A polar bar chart is a circular graph where each axis represents a different variable, typically used for displaying cyclical or directional data. It's suitable for quickly comparing multiple variables across different categories or directions. Let's make one using Vizro-AI.
```

!!! example "Polar Bar Chart"

    === "Code for the cell"
        ```py
        import vizro_ai
        from vizro_ai import VizroAI
        import plotly.express as px

        from dotenv import load_dotenv
        load_dotenv()

        df = px.data.wind()

        vizro_ai = VizroAI(model="gpt-4-0613")
        vizro_ai.plot(df,
                      """Describe wind frequency and direction using bar_polar chart.
                         Increase the width and height of the figure.
                         Improve layout by placing title to the left. Show legend""", explain=True)

        ```
    === "Resulting chart"
        [![VizroAIChart1]][VizroAIChart1]

>     [VizroAIChart1]: ../../assets/user_guides/polar_bar_chart.png

Reviewer 2: Is there a way how to output the same height across all the gifs/screenshots? Eventually we can set it here in the result panel? I just noticed that the surface_plot.gif is quite big compared to the others - is it zoomed in? 
If it's possible to set the same height on the results, that would be great!

Reviewer 3: I don't know how to do it in MkDocs but I'd be tempted to resize them all in an image editor to the same width and upload them in that state.

Author: I'll try to resize them in snaggit. I'm not sure why its so large, the screengrab is actually smaller than the map chart one.

Reviewer 2: That's a good idea! I just don't know if this messes up the aspect ratio, but it would be good if we could consolidate that so it looks a bit cleaner / consistent here 👍

Reviewer 3: You should be able to retain the aspect ratio. I'd aim for 1200 px max width, or even 800 px.

Reviewer 3: All looks good to me, thanks!

> ### Map chart with Vizro-AI

Reviewer 2: ```suggestion
### Create a geographical map chart
```

> ### Map chart with Vizro-AI
> 
> Next chart we are going to explore is a map chart. Map charts allow us to explore and visualize the spatial relationships and patterns in the data, and often provide immediate insights that are not apparent when using other chart types.

Reviewer 2: ```suggestion
### Create a geographical map chart
The next chart we'll look at is a geographical map chart. A geographical map chart often shows a heatmap of values spread out over various regions. It helps us visualize spatial patterns in data, often revealing insights not seen in other charts. 
```

!!! example "Map chart"

    === "Code for the cell"
        ```py
        import vizro_ai
        from vizro_ai import VizroAI
        import plotly.express as px

        from dotenv import load_dotenv
        load_dotenv()

        df = px.data.wind()

        vizro_ai = VizroAI(model="gpt-4-0613")
        vizro_ai.plot(df,
                      """Visualize life expectancy over the years using map chart. Use life expectancy as the color dimension.
                         Improve layout by using Arial font. Increase the width and height of the map area. Outline continents on the map.
                         Show countries on the map.
                         Increase the width and height of the figure.""", explain=True)

        ```
    === "Resulting chart"
        [![VizroAIChart2]][VizroAIChart2]

    [VizroAIChart2]: ../../assets/user_guides/map_chart.gif

> ### Create surface plots with Vizro-AI

Reviewer 2: ```suggestion
### Create a surface plot
```

> With Vizro-AI you can create visually beautiful and engaging charts with a few commands. Let's explore the impressive surface plot VizroAI can generate for us.

Reviewer 2: ```suggestion
With Vizro-AI, you can easily create complex and captivating charts using just a few commands. Let's dive into the 3-dimensional surface plot that VizroAI can generate for us.
```

> With Vizro-AI you can create visually beautiful and engaging charts with a few commands. Let's explore the impressive surface plot VizroAI can generate for us.

Reviewer 3: ```suggestion
Let's explore how to generate the surface plot with Vizro-AI.
```

Reviewer 3: The language is too heavy on marketing.
